zilliqa/src/eth_helpers.rs

```diff
@@ -0,0 +1,62 @@
+use ethabi::{ParamType, Token};
+
+pub(crate) fn extract_revert_msg(encoded: &[u8]) -> String {
+    // It could be either REVERT = keccka256("Error(string)") or PANIC = keccka256("Panic(uint256)")
+    // See https://docs.soliditylang.org/en/latest/control-structures.html#revert
+
+    // Try to decode revert message provided by EVM
+    const REVERT_SELECTOR: [u8; 4] = [0x08, 0xc3, 0x79, 0xa0];
+    const PANIC_SELECTOR: [u8; 4] = [0x4e, 0x48, 0x7b, 0x71];
+
+    let generic_error = String::from("execution reverted");
+    // At least 4 elemenets + anything to decode
+    if encoded.len() < REVERT_SELECTOR.len() + 1 {
+        return generic_error;
+    }
+
+    let payload = &encoded[4..];
+
+    let Ok(vec) = (if encoded.starts_with(&REVERT_SELECTOR) {
```

Comment:
Small nit but I'd prefer to split this over two separate lines to avoid the large bracketed expression. My mind sees the `(`, then needs to scan for the closing `)` and see the `else` after it before I understand what this line does.
i.e.
```rust
let vec = if encoded.starts_with(...) {
  ...
} else {
  ...
};
let Ok(vec) = vec else { return generic_error; };
```